pkg/storage/gcs/store_test.go

```diff
@@ -21,87 +24,111 @@ const (
 	longPath = "this/is/a/long/path/to/an/object/the/object/is/under/this/path/list/with/prefix/please/"
 )
 
-func gen(i int) string {
+func constStringWithIndex(i int) string {
 	return longPath + fmt.Sprint(i)
 }
 
 func setup(t testing.TB, numOfObjects int) (storage.Store, func()) {
 
 	ctx := context.Background()
 
-	bucket := "datamontest-" + internal.RandStringBytesMaskImprSrc(15)
+	bucket := "deleteme-datamontest-" + internal.RandStringBytesMaskImprSrc(15)
 	log.Printf("Created bucket %s ", bucket)
 
 	client, err := gcsStorage.NewClient(context.TODO(), option.WithScopes(gcsStorage.ScopeFullControl))
 	require.NoError(t, err)
 	err = client.Bucket(bucket).Create(ctx, "onec-co", nil)
-	require.NoError(t, err)
+	require.NoError(t, err, "Failed to create bucket:"+bucket)
 
 	gcs, err := New(context.TODO(), bucket, "") // Use GOOGLE_APPLICATION_CREDENTIALS env variable
-	require.NoError(t, err)
+	require.NoError(t, err, "failed to create gcs client")
+	wg := sync.WaitGroup{}
+	create := func(i int, wg *sync.WaitGroup) {
+		err = gcs.Put(ctx, constStringWithIndex(i), bytes.NewBufferString(constStringWithIndex(i)), storage.NoOverWrite)
+		require.NoError(t, err, "Index at: "+fmt.Sprint(i))
+		wg.Done()
```

Comment:
`defer wg.Done()`  more idiomatic and more correct when require exits upon error